supertokens_python/session/session_recipe.py

```diff
@@ -0,0 +1,247 @@
+            })
+        return self.handshake_info
+
+    def update_jwt_signing_public_key_info(self, new_key, new_expiry):
+        if self.handshake_info is not None:
+            self.handshake_info.update_jwt_signing_public_key_info(new_key, new_expiry)
+
+    async def create_new_session(self, request: BaseRequest, user_id: str, jwt_payload: Union[dict, None] = None,
+                                 session_data: Union[dict, None] = None) -> Session:
+        session = await session_functions.create_new_session(self, user_id, jwt_payload, session_data)
+        access_token = session['accessToken']
+        refresh_token = session['refreshToken']
+        id_refresh_token = session['idRefreshToken']
+        new_session = Session(self, access_token['token'], session['session']['handle'],
+                                            session['session']['userId'], session['session']['userDataInJWT'])
+        new_session.new_access_token_info = access_token
+        new_session.new_refresh_token_info = refresh_token
+        new_session.new_id_refresh_token_info = id_refresh_token
+        if 'antiCsrfToken' in session and session['antiCsrfToken'] is not None:
+            new_session.new_anti_csrf_token = session['antiCsrfToken']
+        request.set_session(new_session)
+        return request.get_session()
+
+    async def get_session(self, request: BaseRequest, anti_csrf_check: Union[bool, None] = None, session_required: bool = True) -> Union[Session, None]:
+        id_refresh_token = get_id_refresh_token_from_cookie(request)
+        if id_refresh_token is None:
+            if not session_required:
+                return None
+            raise_unauthorised_exception(self, 'Session does not exist. Are you sending the session tokens in the '
+                                               'request as cookies?')
+        access_token = get_access_token_from_cookie(request)
+        if access_token is None:
+            raise_try_refresh_token_exception(self, 'Access token has expired. Please call the refresh API')
+        anti_csrf_token = get_anti_csrf_header(request)
+        if anti_csrf_check is None:
+            anti_csrf_check = request.method.lower() != 'get'
+        new_session = await session_functions.get_session(self, access_token, anti_csrf_token, anti_csrf_check,
+                                                          get_rid_header(request) is not None)
+        if 'accessToken' in new_session:
+            access_token = new_session['accessToken']['token']
+
+        session = Session(self, access_token, new_session['session']['handle'],
+                                            new_session['session']['userId'], new_session['session']['userDataInJWT'])
+
+        if 'accessToken' in new_session:
+            session.new_access_token_info = new_session['accessToken']
+        request.set_session(session)
+        return request.get_session()
+
+    async def refresh_session(self, request: BaseRequest) -> Session:
```

Comment:
this is a use facing function which means it can be directly used by the user in their API handler. We shouldn't expect them to wrap the request based onto their framework. Rather check here if the request is wrapped or not. if it is not wrapped, then wrap it based on the framework config